packages/grid/_modules_/grid/hooks/features/useApiRef.ts

```diff
@@ -10,14 +10,17 @@ function createGridApi(): GridApi {
   return new EventEmitter() as GridApi;
 }
 
-export function useApiRef(apiRefProp?: ApiRef): ApiRef {
-  const apiRef = React.useRef<GridApi>(createGridApi());
+// Public developers facing overload
+export function useApiRef(): ApiRef;
 
-  React.useEffect(() => {
-    if (apiRefProp?.current) {
-      apiRefProp.current = apiRef.current;
-    }
-  });
+// Internal grid facing overload
+export function useApiRef(apiRefProp: ApiRef | undefined): ApiRef;
```

Comment:
I don't understand why we need an overload here, instead of an optional arg

Reply:
There are two different and distinct use cases: internal, and user-facing. Ideally, I think that we should create two different modules